src/index.ts

```diff
@@ -41,7 +41,14 @@ export const tall = (url: string, options?: Partial<TallOptions>): Promise<strin
           )
         }
 
-        resolve(url)
+        const body = []
+        response.on('data', chunk => body.push(chunk))
+        response.on('end', () => {
+          // Read URL from <meta http-equiv="refresh" /> HTML redirects
+          const metaHttpEquivUrl = body.join('').match(/meta +http-equiv="refresh" +content="\d;url=(http[^"]+)"/)?.[1]
```

Comment:
I wouldn't go for a regex here. What if `http-equiv` and `content` are provided in reverse order or what if there are other attributes between them...

We should be using a library like `cheerio`, [`htmlparser2`](https://github.com/fb55/htmlparser2) or [`parse5`](https://github.com/inikulin/parse5).

Even better if we can efficiently parse tags incrementally (without allocating the entire DOM in memory) and maybe limit the scan only to the `<head></head>` section...

This goes back to my comment in #42 where I suggest we find a "pluggable" way to introduce additional third party dependencies

Reply:
> What if `http-equiv` and `content` are provided in reverse order

Sure, it's definitely naïve :) 

I do think the regex approach may be able to work with some small adaptations, but then we would have to save the DOM to a variable, which is what it sounds like you want to avoid.

> if we can efficiently parse tags incrementally (without allocating the entire DOM in memory)

Hm... building a mini parser... do you have suggestions for what you would like to see in such a thing? Actually iterating over parts of chunks similar to [internals of libraries](https://github.com/mfahlandt/node-xml-stream-parser/blob/4b8a6ffc87d47a700160d3f264dd20b8d569ba16/src/parser.js#L120-L134)?

> and maybe limit the scan only to the `<head></head>` section...

Seems like it could be nice, but HTML without a body but still a single `<meta>` still does the redirect, so maybe we should just look for the `meta` tag.

Reply:
Yeah interesting, the plugin system that you have there. Seems good I guess!

Would you make this `<meta http-equiv />` thing also a default-enabled plugin? Or, alternatively, export it from the codebase, and then add some code examples in the docs so that users can easily enable it without installing another package?

Reply:
Yeah interesting, the plugin system that you have there. Seems ok I guess!

I suppose I would tend away from creating a plugin system with the software that I create, but I understand your points why you want to do it.

Would you make this `<meta http-equiv />` thing also a default-enabled plugin? Or, alternatively, export it from the codebase, and then add some code examples in the docs so that users can easily enable it without installing another package?